packages/zoe/src/zoeService/purseStorage.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+
+import { amountMath } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { makeWeakStore as makeNonVOWeakStore } from '@agoric/store';
+
+import { arrayToObj, objectMap } from '../objArrayConversion';
+
+export const makePurseStorage = () => {
+  /** @type {WeakStore<Brand, ERef<Purse>>} */
+  const brandToPurse = makeNonVOWeakStore('brand');
+
+  /**
+   * Create a purse for a new issuer
+   *
+   * @param {IssuerRecord} record
+   * @returns {void}
+   */
+  const createPurse = record => {
+    if (!brandToPurse.has(record.brand)) {
+      brandToPurse.init(record.brand, E(record.issuer).makeEmptyPurse());
```

Comment:
@michaelfig , given our recent terminology, would it make sense to have a parameterized `Remotable` type so that `T` is a subtype of `Remotable<T>` and `Remotable<T>` is a subtype of `ERef<T>`? A `Remotable<T>` would be defined as "A (primary or remote) remotable whose primary is a `T`".

Reply:
> A (primary or remote) remotable whose primary is a `T`".

I'd like to do that but haven't quite figured out how.  I'll think about it.

Reply:
> * @property {Issuer} issuer which would seem to be correct only if the issuer were always local.

I think this is a big departure from how we've been typing things. I have no assumption that type `Issuer` is always local. So far it has meant either an object that originated in that vat, or a proxy representing an object in another vat. Either fit the type `Issuer`. ERef is typed as `ERef<T> = T | PromiseLike<T>` which just means it could be a promise for the T. It doesn't mean that T is necessarily a proxy. 

We really do not want to type this `issuer` as `ERef<Issuer>` as the purpose of storing the issuer is to ensure it's not a promise for an issuer and that all of its associated data (brand, mathKind, etc) is also available synchronously! So ERef is specifically what we do not want here.

Reply:
Given @michaelfig response https://github.com/Agoric/agoric-sdk/pull/2883#discussion_r613730695 above re `Remotable<T>` it is fine to ignore my suggested change in this PR as we await a better typing. Resolving.